### PR TITLE
[posix] relocate common platform/linux files to platform/posix

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Relocates common posix files from platform/linux to platform/posix

2nd commit removes linux platform from ios building.
Will want to test on other posix platforms (freebsd, linux, android, etc) as the "PlatformDefs.h" changes may cause issues. It shouldnt affect them, but i cant test them all.

Not sure how to handle the PlatformDefs.h. its common for posix, but its pulled from the system_platform dir (win32, linux, etc). All of the *nix variants (freebsd, android, linux) set platform_dir to platform/linux, and therefore the "PlatformDefs.h" header is pulled from there without a full implicit path required as far as i can tell.

## Motivation and Context
https://github.com/xbmc/xbmc/pull/16098

To further the objective of distinct platforms

## How Has This Been Tested?
built and tested on an ipad air running 12.2

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by lrusak** [comment]
Thank you for doing this! I'll give it a test and a review asap!

File: xbmc/platform/darwin/ios/PlatformDefs.h [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+#define PIXEL_ASHIFT 24
+#define PIXEL_RSHIFT 16
+#define PIXEL_GSHIFT 8
+#define PIXEL_BSHIFT 0
+#endif
+
+#include <stdint.h>
+
+#ifndef PRId64
+#ifdef TARGET_WINDOWS
```
Comment (lrusak) [inline xbmc/platform/darwin/ios/PlatformDefs.h]: This is probably just a copy and paste but I hope windows and other platforms don't need access to this file ;)

**Review by lrusak — COMMENTED** [review]
Overall it looks good. We should probably rename some of the linux specific files to use posix (LinuxTimezone, LinuxResourceCounter, etc).

In the future we will also need to separate `CNetworkInterfaceLinux`. I can do this in a follow up PR.

I see you didn't touch OSX, do you need help there or can you include that change also?

**Comment by lrusak** [comment]
jenkins build this please

**Comment by fuzzard** [comment]
I've only done anything for iOS at this stage, as anything around that touches the Apple tv work we have been doing.
I can look at the OSX stuff, but if anyone can jump in, that would be preferred.

And yes, that ifdef is a straight C&P of the platform/Linux/PlatformDefs.h. I can remove the ifdef windows stuff, but mayb it should be ripped out of the Linux copy as well.

**Comment by fuzzard** [comment]
@lrusak Ive moved linx/network out to posix, and have remove all platform/linux from osx.

Tried a full build on osx, but had some issues related to python, not sure if related, so pushed here to see what jenkins says

**Comment by lrusak** [comment]
@fuzzard thank I guess that's fine for now. Ideally I would like to split `CNetworkLinux` into platform specific classes. We can do this in a later PR.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by fuzzard** [comment]
K, see some build failures, ill fix them when the current jenkins pass finishes.

Happy to let someone else take the reins for the CNetworkLinux split. Ill look into the darwin stuff.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by fuzzard** [comment]
All greens on jenkins, and with one last look through, i think thats the last needed for this PR.
Ill wait for the go ahead from someone on the squash for the fixups

File: xbmc/platform/posix/PosixMountProvider.cpp [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 #include <cstdlib>
 
-#include "PosixMountProvider.h"
+#include "platform/posix/PosixMountProvider.h"
```
Comment (lrusak) [inline xbmc/platform/posix/PosixMountProvider.cpp]: this change isn't needed :smiley_cat:

**Comment by lrusak** [comment]
I had a look over and I'm happy. There are some places that I think the headers can be reorganised and cleaned up but I'm ok saving that for later as this PR already has a lot of changes :+1: 

Great job!

**Comment by fuzzard** [comment]
cleaned up, fixed that minor you mentioned. Cheers for the jenkins assistance throughout.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by Rechi** [comment]
I would keep a common PlatformDefs.h file (in posix folder). In the long run we want to get rid of the whole file and duplicating it would complicate the process.

**Comment by fuzzard** [comment]
Would that then not require to set the cmake PLATFORM_DIR to posix for everything (osx/ios/freebsd/android/liinux)?

The initial intent was just to remove the darwin dependency on linux platform for the CCpuInfo PR lrusak had submitted.

I dont really know whats going to happen (if anything) with that  platform_dir change. If the intent is to remove it completely, id say this is the least invasive change for now, as i can test the platforms that are affected here (ios/osx). I understand its early, so big breaking changes are ok, but i dont have android/freebsd/linux devices to do any building/testing with, and at a guess i dont think anyone else has the time to devote to this at the moment.

Your call if you want it done, but will probably need someone to test/check those platforms

**Comment by fuzzard** [comment]
I cant think of an easy way to include PlatformDefs.h from platform/posix without setting the PLATFORM_DIR for each affected platform to platform/posix.

In CMakeLists.cmake in root, headers are included using        
```
set(INCLUDES ${CMAKE_SOURCE_DIR}
             ${CMAKE_SOURCE_DIR}/lib
             ${CMAKE_SOURCE_DIR}/lib/gtest/include
             ${CMAKE_SOURCE_DIR}/xbmc
             ${CMAKE_SOURCE_DIR}/xbmc/${PLATFORM_DIR}
             ${CMAKE_SOURCE_DIR}/xbmc/cores/VideoPlayer
             ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR})
```     
which includes PlatformDefs.h from platform/win32 for windowsstore/windows and all nix flavours currently from platform/linux. 

Unless you add a separate
```
list(APPEND INCLUDES ${CMAKE_SOURCE_DIR}/xbmc/platform/posix)
```
for xbmc/platform/posix for any *nix flavour platform incased in some check for linux platform check?

Is there a macro that can be used to add an include from the ArchSetup.cmake or something?
Just doing a list(APPEND...) from archsetup isnt possible it seems.

Do i create a a new variable to accomodate PlatformDefs maybe that is set in ArchSetup?
Maybe a specific folder for PlatformDefs to not include the same folder twice?

Archsetup.cmake
```
set(PLATFORM_DEFS_DIR platform/posix)
```
CMakeLists.cmake
```
set(INCLUDES ${CMAKE_SOURCE_DIR}
             ${CMAKE_SOURCE_DIR}/lib
             ${CMAKE_SOURCE_DIR}/lib/gtest/include
             ${CMAKE_SOURCE_DIR}/xbmc
             ${CMAKE_SOURCE_DIR}/xbmc/${PLATFORM_DIR}
             ${CMAKE_SOURCE_DIR}/xbmc/${PLATFORM_DEFS_DIR}
             ${CMAKE_SOURCE_DIR}/xbmc/cores/VideoPlayer
             ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR})
```

Any thoughts on best direction?

**Comment by fuzzard** [comment]
Last commit is a possible solution @Rechi

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by fuzzard** [comment]
fixed the freebsd failure. 
The android failure seems to compile/build everything and then fails with some gradle missing thread or something. Not sure if its caused by this, or just a crash on the build server.

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by wsnipex** [comment]
setting PLATFORM_DIR in Archsetup removes the need for includes with platform/[posix|linux..]/foobar.h
You can just use #include "foobar.h"

**Comment by fuzzard** [comment]
> setting PLATFORM_DIR in Archsetup removes the need for includes with platform/[posix|linux..]/foobar.h
> You can just use #include "foobar.h"

This would then require every platform to just define posix then. There would be no distinction between android, darwin, linux, etc.
From what i remember when i looked, i think it was some addon cmake that used the platform_dir var to pull the defines.txt from cmake/platform/xxxx
But also the includes are pulled in from              
```
set(INCLUDES ${CMAKE_SOURCE_DIR}
             ${CMAKE_SOURCE_DIR}/lib
             ${CMAKE_SOURCE_DIR}/lib/gtest/include
             ${CMAKE_SOURCE_DIR}/xbmc
             ${CMAKE_SOURCE_DIR}/xbmc/${PLATFORM_DIR}
             ${CMAKE_SOURCE_DIR}/xbmc/cores/VideoPlayer
             ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR})
```

which would then be incorrect for actual platforms.

**Comment by lrusak** [comment]
> > setting PLATFORM_DIR in Archsetup removes the need for includes with platform/[posix|linux..]/foobar.h
> > You can just use #include "foobar.h"
> 
> This would then require every platform to just define platform_dir as platform/posix then. There would be no distinction between android, darwin, linux, etc. i believe lrusak said the intent was to split all the platforms to their own, so nothing is reliant on another platforms code dirs. Id imagine the intent is to then be assured because you change something that works on linux, that then goes pear shaped because darwin or android are reliant in some undocumented manor.

I see this as a stop gap solution that will be fixed in the future anyways. We don't need a perfect solution right now. The sooner we get this merged the sooner we can move on to splitting up the platforms.

I am fine with this PR as is and thank @fuzzard for the patience an hard work they put into it. :+1:

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
@fuzzard please rebase this at your earliest convenience 🥇

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Review by lrusak — APPROVED** [review]

**Pull request merged**